src/derivative_operators/derivative_operator.jl

```diff
@@ -24,16 +24,25 @@ function CenteredDifference{N}(derivative_order::Int,
     stencil_length          = derivative_order + approximation_order - 1 + (derivative_order+approximation_order)%2
     boundary_stencil_length = derivative_order + approximation_order
     dummy_x                 = -div(stencil_length,2) : div(stencil_length,2)
-    boundary_x              = -boundary_stencil_length+1:0
     boundary_point_count    = div(stencil_length,2) - 1 # -1 due to the ghost point
+
+    left_boundary_x         = 0:(boundary_stencil_length-1)
```

Comment:
why is the centered difference operator changed in this PR?

Reply:
This was a part of the fix of #145 when I merged that branch. Without this other tests will break.

Reply:
Why would other tests break if they weren't broken before?

Reply:
Well they were not testing the required things... like the new error_benchmark test which Alex wrote or the mistake in the stencil which was not summing upto 0. I don't know how but the `ldiv!` tests were passing before.

Reply:
What does this have to do with upwinding?

Reply:
It was a part of#145 which got merged. I merged it as it had changes relevant to upwind operators also.

Reply:
can we have upwinding go in without all of this, and separate it out so I can see the diff? There still hasn't been an upwinding PR...

Reply:
Ok.